### ISSUE TITLE
Proton Drive: error 422 due to authenticating using an empty access token?

### ISSUE THREAD

**Issue by Dzieni** [rclone/rclone#7864]
#### What is the problem you are having with rclone?

Can't access Proton Drive. I get an HTTP error 422.

#### What is your rclone version (output from `rclone version`)

```
michal@dzieni ~ % rclone version
rclone v1.66.0
- os/version: darwin 14.3.1 (64 bit)
- os/kernel: 23.3.0 (arm64)
- os/type: darwin
- os/arch: arm64 (ARMv8 compatible)
- go/version: go1.22.1
- go/linking: dynamic
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

macOS 14.3.1 (23D60), 64-bit

#### Which cloud storage system are you using? (e.g. Google Drive)

Proton Drive

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

```
rclone lsd proton:
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
michal@dzieni ~ % rclone lsd proton: -vv           
2024/05/23 17:06:37 DEBUG : rclone: Version "v1.66.0" starting with parameters ["rclone" "lsd" "proton:" "-vv"]
2024/05/23 17:06:37 DEBUG : Creating backend with remote "proton:"
2024/05/23 17:06:37 DEBUG : Using config file from "/Users/michal/.config/rclone/rclone.conf"
2024/05/23 17:06:37 DEBUG : proton drive root link ID '': Has cached credentials
2024/05/23 17:06:37.378288 WARN RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401), Attempt 1
2024/05/23 17:06:37.378431 ERROR RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401)
2024/05/23 17:06:37.484298 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422), Attempt 1
2024/05/23 17:06:37.484350 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422)
2024/05/23 17:06:37 DEBUG : Saving config "client_uid" in section "proton" of the config file
2024/05/23 17:06:37 DEBUG : Saving config "client_access_token" in section "proton" of the config file
2024/05/23 17:06:37 DEBUG : Saving config "client_refresh_token" in section "proton" of the config file
2024/05/23 17:06:37 DEBUG : Saving config "client_salted_key_pass" in section "proton" of the config file
2024/05/23 17:06:37 DEBUG : proton drive root link ID '': Cached credential doesn't work, clearing and using the fallback login method
2024/05/23 17:06:37 DEBUG : Saving config "client_uid" in section "proton" of the config file
2024/05/23 17:06:37 DEBUG : Saving config "client_access_token" in section "proton" of the config file
2024/05/23 17:06:37 DEBUG : Saving config "client_refresh_token" in section "proton" of the config file
2024/05/23 17:06:37 DEBUG : Saving config "client_salted_key_pass" in section "proton" of the config file
2024/05/23 17:06:37 DEBUG : proton drive root link ID '': couldn't initialize a new proton drive instance using cached credentials: failed to refresh auth: failed to refresh auth, de-auth: 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422)
2024/05/23 17:06:37 DEBUG : proton drive root link ID '': Using username and password to log in
2024/05/23 17:06:38.566218 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422), Attempt 1
2024/05/23 17:06:38.566270 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
2024/05/23 17:06:38 Failed to create file system for "proton:": couldn't initialize a new proton drive instance: 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
```

#### Thoughts

Here is a config for this remote:

```
[proton]
type = protondrive
username = SECRET
password = SECRET
```

I think the problem is following:

- The app believes that there are cached OAuth credentials (which is not true) and tries to authenticate with empty values.
- The request gets rejected (401).
- The app falls back to username & password auth.
- It doesn't work though, because the previous attempts lead to marking the traffic as suspicious.

I believe if the app had started immediately with username & password auth, it would be just fine.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by Dzieni** [comment]
OK, I've tested my hypothesis by adding extra checks for emptiness and unfortunately that's not it. The fixed code immediately tries to log in using username & password, but it does not help with error 422.

```
michal@dzieni rclone % ./rclone lsd proton: -vv
2024/05/23 20:24:33 DEBUG : rclone: Version "v1.67.0-DEV" starting with parameters ["./rclone" "lsd" "proton:" "-vv"]
2024/05/23 20:24:33 DEBUG : Creating backend with remote "proton:"
2024/05/23 20:24:33 DEBUG : Using config file from "/Users/michal/.config/rclone/rclone.conf"
2024/05/23 20:24:33 DEBUG : proton drive root link ID '': Using username and password to log in
2024/05/23 20:24:35.517899 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422), Attempt 1
2024/05/23 20:24:35.517965 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
2024/05/23 20:24:35 Failed to create file system for "proton:": couldn't initialize a new proton drive instance: 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
```

Not sure what Proton means by "traffic coming from your network", but I tried changing my IP address using Proton VPN and it didn't help.

**Comment by henrybear327** [comment]
I think your account has been temporarily jailed / rate limited. Just need to try again tomorrow! :(

**Comment by bestlinuxgamers** [comment]
I got the same problem at basically the same time. I could imagine that it is Proton blocking rclone. I would be interested to know if anyone has managed to log in with rclone in the last time.

**Comment by Dzieni** [comment]
I've contacted Proton using URL received in the error and that's their response:

> Hello,
> 
> Thank you for your message.
> 
> Please note that this error message shows when we detect suspicious or abusive behavior from a specific network. This is part of our anti-abuse measures meant to protect the service and our users' accounts from abusers. 
> 
> **If you are using Rclone, please refrain from using it, this should help solve the issue.** If you are using a VPN service, we suggest connecting to a different server or disabling the connection temporarily.
> 
> Please refer to the following article for a more technical explanation of the issue and for further troubleshooting steps:
> https://proton.me/support/log-in-temporarily-blocked
> 
> Kind regards,
> 
> Riley
> Customer Support
> Proton Abuse Team

Does that mean that Proton offically bans rclone? 😞

**Comment by prankstr** [comment]
Feels like Proton might be blocking rclone based on the [app version](https://rclone.org/protondrive/#protondrive-app-version). I'm running daily backups with pure rclone as well as my own implementation done with https://github.com/henrybear327/Proton-API-Bridge, both has been failing for the past 3 days.

If I keep the original app version of `macos-drive@1.0.0-alpha.1+rclone` I get blocked with `We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins.` etc

If I remove `+proton` in the app version I can immediately login from both rclone and my own implementation.

**Comment by thawthorn** [comment]
> If I remove `+proton` in the app version I can immediately login from both rclone and my own implementation.

I think you mean removing `+rclone` from the app version maybe? This worked for me:

```
rclone lsd protondrive: --protondrive-app-version "macos-drive@1.0.0-alpha.1"
```

Thanks for figuring this out!

**Comment by prankstr** [comment]
@thawthorn Yeah! I thought that's what I said, sorry if I worded it badly :P Glad it worked for you as well

**Comment by EnneGiKVis96** [comment]
I solved the problem by using this: --protondrive-app-version "macos-drive@1.0.0-alpha.1"
as suggested.

The second problem now, at least for me, is that i can't upload files. Rclone says that the proton client is outdated in the logs. Anyone else?

**Comment by Dzieni** [comment]
@EnneGiKVis96 yes, I confirm.

```
michal@dzieni rclone % rclone copy README.md proton:/ --protondrive-app-version "macos-drive@1.0.0-alpha.1"
2024/05/29 10:36:31.621072 WARN RESTY 400 POST https://mail.proton.me/api/drive/blocks: You are using an outdated version of the app. Please update to upload this file’ (Code=2000, Status=400), Attempt 1
```

Most likely `macos-drive@1.0.0-alpha.1` sounds outdated indeed. The solution is probably to log in using a new version of the app, figure out the app version string in the [security logs](https://account.proton.me/u/0/mail/security), and use that value as a `--protondrive-app-version` parameter.

In meanwhile, I've received a reponse from Proton:

> Hello Michał,
>  
> Thank you for your response.
>  
> Please note that Rclone is incompatible with our service, so using it will generate multiple unsuccessful logs, which triggers our security system. 
>  
> Due to the reason stated above, please refrain from using Rclone to prevent such login alerts.
> 
> Kind regards,
> 
> Riley
> Customer Support
> Proton Abuse Team

So it looks like the only way to work with Proton is to emulate their official client, which means being always one step behind. Not sure what is the point of sharing [an official SDK](https://github.com/ProtonMail/go-proton-api) if you cannot use it without hacks. Deeply disappointing, especially considering the customer base of Proton - a tech-savvy community eagerly willing to help make the product more useful.

I think that's enough for me to change my mind about keeping the subscription.

**Comment by prankstr** [comment]
Just changing the app version won't do much in this case unfortunately. The API bridge used by rclone to interact with proton most likely have to be updated to accommodate changes to the Proton backend.

I'm reconsidering my subscription as well :/

**Comment by EnneGiKVis96** [comment]
i think this already happened before, because i have this exact problem i think 1 month ago, and rclone resolve it.
It's a shame that Proton don't like open source solutions that cover their lack of support for the linux community tho.

**Comment by SamSphere** [comment]
> Just changing the app version won't do much in this case unfortunately. The API bridge used by rclone to interact with proton most likely have to be updated to accommodate changes to the Proton backend.
> 
> I'm reconsidering my subscription as well :/

Dropping the app-version tag when uploading seems to works. 
`rclone copy test.pdf proton: --protondrive-replace-existing-draft=true --progress --protondrive-2fa=UseYour6Digit2FA`

**Comment by SamSphere** [comment]
> Just changing the app version won't do much in this case unfortunately. The API bridge used by rclone to interact with proton most likely have to be updated to accommodate changes to the Proton backend.
> 
> I'm reconsidering my subscription as well :/

Dropping the app-version tag when uploading seems to solve the issue.


```shell
rclone copy test.pdf proton: --protondrive-replace-existing-draft=true --progress --protondrive-2fa=UseYour6Digit2FA
```

**Comment by thawthorn** [comment]
I moved to Filen.io and am happy.

**Comment by LivingUserError** [comment]
I have found that this issue a) persists and b) might be much less complicated than it appears.

First of all, having 'rclone' in the app-name string does not seem to cause any problems, or at least not anymore:
![image](https://github.com/user-attachments/assets/84649432-500d-4d7f-a426-1faefa1d60bd)

Secondly, I have found
```
client_uid =
client_access_token =
client_refresh_token =
client_salted_key_pass =
```
in my rclone config after the failed login attempt, which is obviously done by rclone, not Proton.

If there is a connection problem, in my case a simple timeout, the session info is wiped by rclone. I assume this is done to rule out that the problem is with the session, rather than something else. However, this does not work well at all in the case of Proton!

1. If a user has 2FA enabled, wiping the session info means 2FA is required again, so manual intervention is required at this point.
1. It seems that logins through a VPN (surprisingly even ProtonVPN) get blocked by Proton Sentinel. I am not sure why exactly, because for example for Android apps or in a browser it works, but with rclone it does not. This means VPN users have to completely disable VPN before being able to login again.

In sum, I think at leat for Proton, rclone should reconsider the session info wipes. It seriously negatively impacts the user experience of anyone using 2FA (which should be anyone :P) and even more so if you use any VPN.

**Comment by maaaaz** [comment]
This issue is still present, protondrive remote can't be used even in the latest rclone version. 
Any plan to fix it ?

**Comment by LivingUserError** [comment]
I use some fish scripts that backup and restore the config as needed atm:

`pre_sync.fish`
```fish
set curr_id (cat /home/user/.config/rclone/rclone.conf | string match -g -r "client_uid = (.*)")
set old_id (cat /home/user/.config/rclone/rclone.conf.old | string match -g -r "client_uid = (.*)")

if test -z "$curr_id"
    if test -n "$old_id"
        cp /home/user/.config/rclone/rclone.conf.old /home/user/.config/rclone/rclone.conf
    end
else
    cp /home/user/.config/rclone/rclone.conf /home/user/.config/rclone/rclone.conf.old
end
```

`post_sync.fish`
```fish
# Catch exit status and check whether rclone deleted session info (error case)
function post_sync --on-process-exit $last_pid
    set -g sync_status $argv[3]
    if test $argv[3] -eq 0
        cp /home/user/.config/rclone/rclone.conf /home/user/.config/rclone/rclone.conf.old
    else
        set curr_id (cat /home/user/.config/rclone/rclone.conf | string match -g -r "client_uid = (.*)")
        set old_id (cat /home/user/.config/rclone/rclone.conf.old | string match -g -r "client_uid = (.*)")
        if test -n "$old_id"
            if test -z "$curr_id"
                cp /home/user/.config/rclone/rclone.conf.old /home/user/.config/rclone/rclone.conf
            end
        end
    end
end
```

Usage:
```fish
#!/usr/bin/env fish

source /home/user/scripts/helper/pre_sync.fish

# something using rclone or rclone command

source /home/user/scripts/helper/post_sync.fish
```

Adjust paths as needed.

Note that this will only work on Linux (or with WSL, Cygwin) and if you have [fish shell](https://fishshell.com/) installed. If you do, you should be able to execute such a script like any other.

**Comment by fhoekstra** [comment]
@LivingUserError is it possible to use these scripts with a VFS cached mount?

**Comment by LivingUserError** [comment]
@fhoekstra I don't see why not. Basically all they do is take your `.conf` exactly as is and backup/restore it.

`pre_sync.fish` will create a backup of the `.conf` if there is a session ID found. If this is not the case, it will attempt to restore from an existing backup, given there is one and it contains a session ID.

`post_sync.fish` is a bit more complex. It creates a hook that will be executed after whatever background process is started before it (using the `$last_pid` variable that is built into `fish`).

What is does is creating/refreshing your backup if it reads exit code 0. If it detects a non-zero exit, it will test whether your current `.conf` still holds a session ID. If this is not the case, it will look for a backup. If there is one and it contains a session ID, it will restore it.

So far, I haven't experienced any issue while running `restic` backups and `rclone bisync`s with these scripts in place. However, **I want to stress that it's just a quick-n-dirty solution I came up with!** I am not working on this project in any official capacity.

**Comment by behrangsa** [comment]
Just fyi,

For one on my accounts that doesn't have 2FA, it works:

```
❯ rclone copy -v weblinks.txt foo: --protondrive-replace-existing-draft=true -v
<7>DEBUG : rclone: Version "v1.66.0-beta.7550.85f142a20" starting with parameters ["rclone" "copy" "-v" "weblinks.txt" "foo:" "--protondrive-replace-existing-draft=true" "-v"]
<7>DEBUG : rclone: systemd logging support activated
<7>DEBUG : Creating backend with remote "weblinks.txt"
<7>DEBUG : Using config file from "/home/dude/.config/rclone/rclone.conf"
<7>DEBUG : fs cache: adding new entry for parent of "weblinks.txt", "/home/dude/Documents"
<7>DEBUG : Creating backend with remote "foo:"
<7>DEBUG : foo: detected overridden config - adding "{kpshc}" suffix to name
<7>DEBUG : proton drive root link ID '': Has cached credentials
<7>DEBUG : proton drive root link ID '': Used cached credential to initialize the ProtonDrive API
<7>DEBUG : fs cache: renaming cache item "amiga:" to be canonical "amiga{kpshc}:"
<7>DEBUG : weblinks.txt: Size and modification time the same (differ by -728.196702ms, within tolerance 1s)
<7>DEBUG : weblinks.txt: Unchanged skipping
<6>INFO  : 
Transferred:   	          0 B / 0 B, -, 0 B/s, ETA -
Elapsed time:         6.9s
```

For some reason I should add `-v` twice to enable verbose logging.

For my account, with 2FA, it fails:

```
$ rclone copy -v weblinks.txt bar: --protondrive-replace-existing-draft=true -v --protondrive-2fa=477272
<7>DEBUG : rclone: Version "v1.66.0-beta.7550.85f142a20" starting with parameters ["rclone" "copy" "-v" "weblinks.txt" "bar:" "--protondrive-replace-existing-draft=true" "-v" "--protondrive-2fa=477272"]
<7>DEBUG : rclone: systemd logging support activated
<7>DEBUG : Creating backend with remote "weblinks.txt"
<7>DEBUG : Using config file from "/home/dude/.config/rclone/rclone.conf"
<7>DEBUG : fs cache: adding new entry for parent of "weblinks.txt", "/home/dude/Documents"
<7>DEBUG : Creating backend with remote "bar:"
<7>DEBUG : bar: detected overridden config - adding "{BR-F2}" suffix to name
<7>DEBUG : proton drive root link ID '': Has cached credentials
2025/04/18 16:10:41.418578 WARN RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401), Attempt 1
2025/04/18 16:10:41.418648 ERROR RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401)
2025/04/18 16:10:41.831885 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422), Attempt 1
2025/04/18 16:10:41.831923 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422)
<7>DEBUG : Saving config "client_uid" in section "bar" of the config file
<7>DEBUG : Saving config "client_access_token" in section "bar" of the config file
<7>DEBUG : Saving config "client_refresh_token" in section "bar" of the config file
<7>DEBUG : Saving config "client_salted_key_pass" in section "bar" of the config file
<7>DEBUG : proton drive root link ID '': Cached credential doesn't work, clearing and using the fallback login method
<7>DEBUG : Saving config "client_uid" in section "bar" of the config file
<7>DEBUG : Saving config "client_access_token" in section "bar" of the config file
<7>DEBUG : Saving config "client_refresh_token" in section "bar" of the config file
<7>DEBUG : Saving config "client_salted_key_pass" in section "bar" of the config file
<7>DEBUG : proton drive root link ID '': couldn't initialize a new proton drive instance using cached credentials: failed to refresh auth: failed to refresh auth, de-auth: 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422)
<7>DEBUG : proton drive root link ID '': Using username and password to log in
2025/04/18 16:10:43.470496 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422), Attempt 1
2025/04/18 16:10:43.470525 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
Failed to create file system for "bar:": couldn't initialize a new proton drive instance: 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
```

**Comment by LivingUserError** [comment]
> Just fyi,
> 
> For one on my accounts that doesn't have 2FA, it works:
> 
> I have [posted this](https://www.reddit.com/r/ProtonDrive/comments/1k24cey/trouble_working_with_protondrive_using_rclone/) in the official (?) ProtonDrive subreddit too:
> 
> ```
> ❯ rclone copy -v weblinks.txt foo: --protondrive-replace-existing-draft=true -v
> <7>DEBUG : rclone: Version "v1.66.0-beta.7550.85f142a20" starting with parameters ["rclone" "copy" "-v" "weblinks.txt" "foo:" "--protondrive-replace-existing-draft=true" "-v"]
> <7>DEBUG : rclone: systemd logging support activated
> <7>DEBUG : Creating backend with remote "weblinks.txt"
> <7>DEBUG : Using config file from "/home/dude/.config/rclone/rclone.conf"
> <7>DEBUG : fs cache: adding new entry for parent of "weblinks.txt", "/home/dude/Documents"
> <7>DEBUG : Creating backend with remote "foo:"
> <7>DEBUG : foo: detected overridden config - adding "{kpshc}" suffix to name
> <7>DEBUG : proton drive root link ID '': Has cached credentials
> <7>DEBUG : proton drive root link ID '': Used cached credential to initialize the ProtonDrive API
> <7>DEBUG : fs cache: renaming cache item "foo:" to be canonical "foo{kpshc}:"
> <7>DEBUG : weblinks.txt: Size and modification time the same (differ by -728.196702ms, within tolerance 1s)
> <7>DEBUG : weblinks.txt: Unchanged skipping
> <6>INFO  : 
> Transferred:   	          0 B / 0 B, -, 0 B/s, ETA -
> Elapsed time:         6.9s
> ```
> 
> For some reason I should add `-v` twice to enable verbose logging.
> 
> For my account, with 2FA, it fails:
> 
> ```
> $ rclone copy -v weblinks.txt bar: --protondrive-replace-existing-draft=true -v --protondrive-2fa=477272
> <7>DEBUG : rclone: Version "v1.66.0-beta.7550.85f142a20" starting with parameters ["rclone" "copy" "-v" "weblinks.txt" "bar:" "--protondrive-replace-existing-draft=true" "-v" "--protondrive-2fa=477272"]
> <7>DEBUG : rclone: systemd logging support activated
> <7>DEBUG : Creating backend with remote "weblinks.txt"
> <7>DEBUG : Using config file from "/home/dude/.config/rclone/rclone.conf"
> <7>DEBUG : fs cache: adding new entry for parent of "weblinks.txt", "/home/dude/Documents"
> <7>DEBUG : Creating backend with remote "bar:"
> <7>DEBUG : bar: detected overridden config - adding "{BR-F2}" suffix to name
> <7>DEBUG : proton drive root link ID '': Has cached credentials
> 2025/04/18 16:10:41.418578 WARN RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401), Attempt 1
> 2025/04/18 16:10:41.418648 ERROR RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401)
> 2025/04/18 16:10:41.831885 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422), Attempt 1
> 2025/04/18 16:10:41.831923 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422)
> <7>DEBUG : Saving config "client_uid" in section "bar" of the config file
> <7>DEBUG : Saving config "client_access_token" in section "bar" of the config file
> <7>DEBUG : Saving config "client_refresh_token" in section "bar" of the config file
> <7>DEBUG : Saving config "client_salted_key_pass" in section "bar" of the config file
> <7>DEBUG : proton drive root link ID '': Cached credential doesn't work, clearing and using the fallback login method
> <7>DEBUG : Saving config "client_uid" in section "bar" of the config file
> <7>DEBUG : Saving config "client_access_token" in section "bar" of the config file
> <7>DEBUG : Saving config "client_refresh_token" in section "bar" of the config file
> <7>DEBUG : Saving config "client_salted_key_pass" in section "bar" of the config file
> <7>DEBUG : proton drive root link ID '': couldn't initialize a new proton drive instance using cached credentials: failed to refresh auth: failed to refresh auth, de-auth: 422 POST https://mail.proton.me/api/auth/v4/refresh: Invalid refresh token (Code=10013, Status=422)
> <7>DEBUG : proton drive root link ID '': Using username and password to log in
> 2025/04/18 16:10:43.470496 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422), Attempt 1
> 2025/04/18 16:10:43.470525 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
> Failed to create file system for "bar:": couldn't initialize a new proton drive instance: 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
> ```

This means you got IP blocked. Happens sometimes when you use a VPN (including Proton VPN, although I didn't get this error in a while). My suggested workaround can still help, since it allows you to keep your old session info and retry without providing a 2FA code again.

**Comment by vanvuvuong** [comment]
> I have found that this issue a) persists and b) might be much less complicated than it appears.
> 
> First of all, having 'rclone' in the app-name string does not seem to cause any problems, or at least not anymore: ![image](https://github.com/user-attachments/assets/84649432-500d-4d7f-a426-1faefa1d60bd)
> 
> Secondly, I have found
> 
> ```
> client_uid =
> client_access_token =
> client_refresh_token =
> client_salted_key_pass =
> ```
> 
> in my rclone config after the failed login attempt, which is obviously done by rclone, not Proton.
> 
> If there is a connection problem, in my case a simple timeout, the session info is wiped by rclone. I assume this is done to rule out that the problem is with the session, rather than something else. However, this does not work well at all in the case of Proton!
> 
> 1. If a user has 2FA enabled, wiping the session info means 2FA is required again, so manual intervention is required at this point.
> 2. It seems that logins through a VPN (surprisingly even ProtonVPN) get blocked by Proton Sentinel. I am not sure why exactly, because for example for Android apps or in a browser it works, but with rclone it does not. This means VPN users have to completely disable VPN before being able to login again.
> 
> In sum, I think at leat for Proton, rclone should reconsider the session info wipes. It seriously negatively impacts the user experience of anyone using 2FA (which should be anyone :P) and even more so if you use any VPN.

Thanks for this sum-up. I use 2FA and ProtonVPN, it blocked me
```
2025/11/08 10:22:13.630204 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422), Attempt 1
2025/11/08 10:22:13.630249 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4: We are detecting potentially abusive traffic coming from your network and have temporarily blocked logins. If you believe this is in error, please contact us here: https://proton.me/support/appeal-abuse (Code=2028, Status=422)
```
Switch-off the VPN and use flag `--protondrive-2fa`, it worked like a charm.

**Comment by alexeicolin** [comment]
**EDIT: This report was for rclone `1.73.?`. The issue appears to be fixed in `1.74.2` (see next comment).**

Was able to login through ProtonVPN with `--protondrive-2fa=123456` and `app_version = macos-drive@1.0.0-alpha.1` in config file. The session shows up in the Security page in the web interface.

This success is shortly after having gotten the `We are detecting potentially abusive traffic...` error on first attempt (with app_version = macos-drive@1.0.0-alpha.1+rclone` though not confirmed that removing the `+rclone` suffix made the difference, maybe not getting blocked is random luck). Also, after having gotten 'Invalid access token` error (with `macos-drive@1.0.0-alpha.1+rclone` and `android-drive=2.33.0`) -- even on a brand new remote created in rclone (so this was not due to stale access tokens).

Listing directory works (but it is extremely slow, ~4sec per each child directory with just 10 files each, which seems like ~4sec per request.

But, copying files does not work: folders are created successfully but copying of the file errors out with `WARN RESTY 400 POST https://mail.proton.me/api/drive/blocks: You are using an outdated version of the app. Please update to upload this file. (Code=2000, Status=400), Attempt 1`.
**EDIT: Apparently there's a patch for this:** https://github.com/henrybear327/go-proton-api/pull/5

Tried with `--protondrive-app-version=android-drive=2.33.0`, but on same open session (so presumably this parameter does not matter since the session is already authenticated). As someone said above, it's probably the implementation of the client itself in rclone is no longer compatible with Proton server.  The only strange thing is how copy worked for a few commenters above (perhaps things have changed since then).

**Comment by alexeicolin** [comment]
Appears to be fixed in rclone 1.74.2 (Thank you!)
**Have to re-create the remote.**
Copy of files works.

The app version left blank in rclone config, autopopulated by rclone. Shows up as -`external-drive-rclone 1.74.2-stable` in the Proton web interface.

**Comment by ncw** [comment]
Thanks for confirming that @alexeicolin - I will close this issue now.

**Comment by manavchaudhary1** [comment]
I am having same issue
```
> rclone ls proton: -vv --protondrive-2fa=XXXXX
2026/06/25 01:18:42 DEBUG : rclone: Version "v1.74.0" starting with parameters ["rclone" "ls" "proton:" "-vv" "--protondrive-2fa=331143"]
2026/06/25 01:18:42 DEBUG : Creating backend with remote "proton:"
2026/06/25 01:18:42 DEBUG : Using config file from "/home/manav/.config/rclone/rclone.conf"
2026/06/25 01:18:42 DEBUG : proton: detected overridden config - adding "{m3jqg}" suffix to name
2026/06/25 01:18:42 DEBUG : proton drive root link ID '': Using username and password to log in
2026/06/25 01:18:46.044391 WARN RESTY 401 GET https://drive-api.proton.me/core/v4/users: Invalid access token (Code=401, Status=401), Attempt 1
2026/06/25 01:18:46.044407 ERROR RESTY 401 GET https://drive-api.proton.me/core/v4/users: Invalid access token (Code=401, Status=401)
2026/06/25 01:18:46 DEBUG : Saving config "client_uid" in section "proton" of the config file
2026/06/25 01:18:46 DEBUG : Saving config "client_access_token" in section "proton" of the config file
2026/06/25 01:18:46 DEBUG : Saving config "client_refresh_token" in section "proton" of the config file
2026/06/25 01:18:46 DEBUG : Saving config "client_salted_key_pass" in section "proton" of the config file
2026/06/25 01:18:50 DEBUG : proton drive root link ID '': Used username and password to initialize the ProtonDrive API
2026/06/25 01:18:50 DEBUG : Saving config "client_uid" in section "proton" of the config file
2026/06/25 01:18:50 DEBUG : Saving config "client_access_token" in section "proton" of the config file
2026/06/25 01:18:50 DEBUG : Saving config "client_refresh_token" in section "proton" of the config file
2026/06/25 01:18:50 DEBUG : Saving config "client_salted_key_pass" in section "proton" of the config file
2026/06/25 01:18:50 DEBUG : fs cache: renaming cache item "proton:" to be canonical "proton{m3jqg}:"
2026/06/25 01:18:51 DEBUG : LSMS-Guide-20260610: Original file size missing
2026/06/25 01:18:51 DEBUG : LSMS-Guide-20260610: Original file size missing
        0 LSMS-Guide-20260610
```

But then i ran 
```
> rclone ls proton:
2026/06/25 01:19:27.874088 WARN RESTY 401 GET https://drive-api.proton.me/core/v4/users: Invalid access token (Code=401, Status=401), Attempt 1
2026/06/25 01:19:27.874133 ERROR RESTY 401 GET https://drive-api.proton.me/core/v4/users: Invalid access token (Code=401, Status=401)
2026/06/25 01:19:28.336209 WARN RESTY 400 POST https://drive-api.proton.me/auth/v4/refresh: Invalid refresh token (Code=10013, Status=400), Attempt 1
2026/06/25 01:19:28.336223 ERROR RESTY 400 POST https://drive-api.proton.me/auth/v4/refresh: Invalid refresh token (Code=10013, Status=400)
2026/06/25 01:19:29 CRITICAL: Failed to create file system for "proton:": couldn't initialize a new proton drive instance: this account requires a 2FA code. Can be provided with --protondrive-2fa=000000
``` 

before running second cmd
```
❯ cat /home/manav/.config/rclone/rclone.conf
[proton]
type = protondrive
username = somethin
password = somethin
client_uid = somethin
client_access_token = somethin 
client_refresh_token = somethin
client_salted_key_pass = somethin
```

and just then it became empty
```
❯ cat /home/manav/.config/rclone/rclone.conf
[proton]
type = protondrive
username = somethin
password = somethin
client_uid =
client_access_token =
client_refresh_token =
client_salted_key_pass =
```
